sanic_jwt/configuration.py

```diff
@@ -1,119 +1,313 @@
+    "authorization_header_refresh_prefix": "Refresh",
+    "claim_aud": None,
+    "claim_iat": False,
+    "claim_iss": None,
+    "claim_nbf": False,
+    "claim_nbf_delta": 0,
+    "cookie_domain": "",
+    "cookie_httponly": True,
+    "cookie_refresh_token_name": "refresh_token",
+    "cookie_set": False,
+    "cookie_strict": True,
+    "cookie_access_token_name": "access_token",
+    "debug": False,
+    "expiration_delta": 60 * 5 * 6,
+    "generate_refresh_token": utils.generate_token,
+    "leeway": 60 * 3,
+    "refresh_token_enabled": False,
+    "refresh_token_name": "refresh_token",
+    "path_to_authenticate": "/",
+    "path_to_retrieve_user": "/me",
+    "path_to_verify": "/verify",
+    "path_to_refresh": "/refresh",
+    "private_key": None,
+    "scopes_enabled": False,
+    "scopes_name": "scopes",
+    "secret": "This is a big secret. Shhhhh",
+    "strict_slashes": False,
+    "url_prefix": "/auth",
+    "user_id": "user_id",
+    "verify_exp": True
 }
 
 aliases = {
-    'cookie_token_name': 'cookie_access_token_name',
-    'public_key': 'secret',
+    "cookie_access_token_name": "cookie_token_name", "secret": "public_key"
 }
 
+ignore_keys = (
+    "add_scopes_to_payload",
+    "authenticate",
+    "class_views",
+    "extend_payload",
+    "retrieve_refresh_token",
+    "retrieve_user",
+    "store_refresh_token",
+)
+
 logger = logging.getLogger(__name__)
-config = None
+
+
+def _warn_key(key):
+    if key not in ignore_keys:
+        logger.warning(
+            "Configuration key '%s' found is not valid for sanic-jwt", key
+        )
+
+
+def _create_or_overwrite_config_item(value, key, item_aliases, instance):
+    setattr(
+        instance,
+        key,
+        ConfigItem(
+            value, item_name=key, config=instance, aliases=item_aliases
+        ),
+    )
+
+
+def _update_config_item(key, item_aliases, instance):
+    getattr(instance, key)._item_name = key
+    getattr(instance, key)._aliases = item_aliases
+    getattr(instance, key)._config = instance
+
+
+class ConfigItem:
+
+    def __init__(
+        self,
+        value,
+        item_name=None,
+        config=None,
+        inject_request=True,
+        aliases=None,
+    ):
+        self._value = value
+        self._item_name = item_name
+        self._config = config
+        self._inject_request = inject_request
+
+        if aliases is not None and isinstance(aliases, (list, tuple, set)):
+            self._aliases = aliases
+        else:
+            self._aliases = []
+
+    def update(self, value):
+        self._value = value
+
+    def __call__(self, **kwargs):
+        if asyncio.get_event_loop().is_running():
+            if self._get_from_config is not None:
+                args = []
+
+                if self._inject_request and is_cached("_request"):
+                    args.append(get_cached("_request"))
+                return self._get_from_config.__call__(*args)
+
+            if is_cached(self._item_name):
```

Comment:
Shouldn’t we swap the order of these? I was thinking it would be:

1. Check the cache
2. Check the getter method
3. Return the fallback `_value`

If it was not cached and we are executing the getter, then it would execute only once and cache its value when done. Am I missing something?

Reply:
You're right. My fault on that :sweat_smile: